src/renderers/webgl/WebGLProgram.js

```diff
@@ -1053,6 +1053,8 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 	this.vertexShader = glVertexShader;
 	this.fragmentShader = glFragmentShader;
 
+	this.setsPointSize = /gl_PointSize\s*=/.test( vertexShader );
```

Comment:
What is the purpose of this line? Doesn't a points material not always set `gl_PointSize`?

Reply:
Yes, but common override materials like `MeshNormalMaterial` and `MeshDepthMaterial` do not set `gl_PointSize`. We could detect points-compatible materials some other way too, if you think there's another good alternative. This seemed simple but it is a bit hacky, and can't detect whether point size is set within a comment or in the actual code.

Reply:
Sorry, I was reading the code wrong. I'm not feeling good with a check in `renderBufferDirect()`.

How about you check if `scene.overrideMaterial !== null` in `projectObject()` and then don't add points into the render list?

Reply:
Wait, I'm not sure this is right. What if a developer uses a custom, points-compatible shader material for `Scene.overrideMaterial`? Always excluding points from `Scene.overrideMaterial` would break this usage...